sw/applications/example_external_peripheral/main.c

```diff
@@ -25,6 +25,10 @@ plic_irq_id_t intr_num;
 int main(int argc, char *argv[])
 {
 
+#ifndef RV_PLIC_IS_INCLUDED
+  #pragma message ( "This app does NOT work as the RV_PLIC peripheral is not included" )
```

Comment:
I would output an error, so the app is not even compiled. If we do not have the PLIC for example, why do we want the app to be able to be compiled? so:
`#pragma error ( "This app does NOT work as the RV_PLIC peripheral is not included" )`
or
`#error ( "This app does NOT work as the RV_PLIC peripheral is not included" ) `
Both work

Reply:
We have to be careful with this or the CI might reject every push. We might need to compile for both targets and make a logic OR maybe?

Reply:
yes I kept it like this as otherwise the application does not end, maybe it is better to return 0 rather than -1 as indeed the CI may complain that application which is not supposed to run fail

Reply:
Having talked with Jose we agree that an error would be a good solution.

Reply:
and what about the CI?

Reply:
so, we want the application to be able to run even though we do not have (for example) the PLIC included?

Reply:
no but the CI compiles anyway all the program and a future CI task may run all the applications, so we need to find a way that the CI does not report "errors" -

Reply:
But if I push an application that requires a change in the `mcu_cfg.hjson` to compile, I think the CI should raise an error.

Reply:
unless i am missing something, I agree

Reply:
done @JuanSapriza @JoseCalero